core/src/main/java/org/apache/calcite/rel/metadata/BuiltInMetadata.java

```diff
@@ -105,6 +105,44 @@ interface Handler extends MetadataHandler<UniqueKeys> {
     }
   }
 
+  /**
+   * Metadata about which columns have foreign keys.
+   */
+  public interface ForeignKeys extends Metadata {
+    MetadataDef<ForeignKeys> DEF =
+        MetadataDef.of(ForeignKeys.class, ForeignKeys.Handler.class,
+            BuiltInMethod.FOREIGN_KEYS.method);
+
+    /**
+     * Determines the list of foreign keys for this expression. Foreign keys are
+     * represented as an {@link org.apache.calcite.util.ImmutableBitSet}, where
+     * each bit position represents the column ordinal is foreign key.
+     *
+     * @param ignoreNulls if true, ignore null values when determining
+     *                    whether the keys are foreign keys
+     *
+     * @return set of foreign keys, or empty set if this information
+     * cannot be determined (whereas null indicates definitely
```

Comment:
Can you sketch an example for the "empty vs null" case here?

Reply:
I make a mistake. It should return empty ImmutableBitSet but never null If the current relNode(mainly underlying tableScan relNode correspondingly) does't contains foreign keys. I will fix it.